src/main/java/com/theelixrlabs/healthcare/service/DoctorService.java

```diff
@@ -1,37 +1,59 @@
 package com.theelixrlabs.healthcare.service;
 
 import com.theelixrlabs.healthcare.constants.DoctorConstants;
+import com.theelixrlabs.healthcare.exceptionHandler.CustomException;
 import com.theelixrlabs.healthcare.modal.DoctorEntity;
 import com.theelixrlabs.healthcare.repository.DoctorRepository;
 import com.theelixrlabs.healthcare.dto.DoctorDto;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.UUID;
+
 /**
  * Service class responsible for handling operations related to doctor.
  */
 @Service
 public class DoctorService {
-    @Autowired
-    private DoctorRepository doctorRepository;
+    private final DoctorRepository doctorRepository;
+
+    public DoctorService(DoctorRepository doctorRepository) {
+        this.doctorRepository = doctorRepository;
+    }
 
     /**
      * Saves a new doctor document based on the provided DoctorDto.
      *
      * @param doctorDto : DoctorDto object containing doctor information.
      * @return The saved dto object.
      */
-    public DoctorDto saveDoctor(DoctorDto doctorDto) throws RuntimeException {
-        if (doctorRepository.findByAadhaar(doctorDto.getAadhaar()).isPresent()) {
-            throw new RuntimeException(DoctorConstants.AADHAAR_ALREADY_PRESENT);
+    public DoctorDto saveDoctor(DoctorDto doctorDto) throws CustomException {
+        validateDoctor(doctorDto);
+        if (doctorRepository.findByAadhaarNumber(doctorDto.getAadhaarNumber()).isPresent()) {
+            throw new CustomException(DoctorConstants.AADHAAR_ALREADY_PRESENT);
         }
-        DoctorEntity doctorEntity = new DoctorEntity();
-        doctorEntity.setFirstName(doctorDto.getFirstName());
-        doctorEntity.setLastName(doctorDto.getLastName());
-        doctorEntity.setDepartment(doctorDto.getDepartment());
-        doctorEntity.setAadhaar(doctorDto.getAadhaar());
+        UUID uuid = UUID.randomUUID();
+        DoctorEntity doctorEntity = DoctorEntity.builder().id(uuid).firstName(doctorDto.getFirstName()).lastName(doctorDto.getLastName()).department(doctorDto.getDepartment()).aadhaarNumber(doctorDto.getAadhaarNumber()).build();
         doctorRepository.save(doctorEntity);
-        doctorDto.setId(doctorEntity.getId());
-        return doctorDto;
+        DoctorDto savedDoctorDto;
+        savedDoctorDto = DoctorDto.builder().id(doctorEntity.getId()).firstName(doctorEntity.getFirstName()).lastName(doctorEntity.getLastName()).department(doctorEntity.getDepartment()).aadhaarNumber(doctorEntity.getAadhaarNumber()).build();
```

Comment:
line 35 and 38 split into multiple lines 
make it readable

Reply:
Okay. Changed